python/test/test_publisher.py

```diff
@@ -1448,6 +1451,25 @@ def test_publish_json(self):
             }
             gha.set_output.assert_called_once_with('json', json.dumps(expected))
 
+    def test_publish_summary(self):
+        with tempfile.TemporaryDirectory() as path:
+            filepath = os.path.join(path, 'summary.md')
+            settings = self.create_settings(job_summary=True)
+            gh, gha, req, repo, commit = self.create_mocks(digest=self.base_digest, check_names=[settings.check_name])
+            publisher = Publisher(settings, gh, gha)
+
+            title = "A title"
+            summary = 'Well, this summary is not very interesting'
+            
+            with patch.dict(os.environ, {'GITHUB_STEP_SUMMARY': filepath}):
+                publisher.publish_summary(title, summary)
+
+            with open(filepath, 'r') as summary_file:
+                contents = summary_file.read()
+                self.assertIn(title, contents, 'Title is not in the summary')
+                self.assertIn(summary, contents, 'Summary is not in the summary file')
```

Comment:
```suggestion
                self.assertEqual(contents, '## Title is not in the summary\nSummary is not in the summary file')
```

Reply:
I think you misread the assertions here: the 3rd argument is the message in case assertion failed, the title & summary are expressed in variables before.
My goal is to write test as a contract that title & summary are present in the file, regardless of the formatting around which may be adjusted without real impact. Otherwise you'll see your tests maintenance growing as a burden. That's why `assertIn` is generally enough (even though Github copilot didn't generate the same message for both assertions, which I'm gonna fix)

Reply:
I am aware of that burden. I prefer to have at least one test that asserts the entire summary to have something flagging unwanted changes sneaking into the summary. And this tests looks like the right place for it.